### PR TITLE
Setters now check for null

### PR DESCRIPTION
**Pull request by alisonreynoso**
## Summary
- Model's setters check for null
- Added guava's precondition dependendency
## Trello Card
https://trello.com/c/xQP6Rjbi/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

**Review by Jimenarosello — COMMENTED** [review]
Nice job!! 😄  I recommend checking out the `checkArgument` method that allows to validate other things besides null fields, maybe you could add some validations with that method as well :)

**Comment by julianvasquez-wolox** [comment]
As Jime said, you should apply some extra validations using checkArguments ( like no empty values or the year not to be greater than the actual year ) 😊

**Review by julianvasquez-wolox — CHANGES_REQUESTED** [review]
Check the comments 😁

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -68,54 +75,57 @@ public String getImage() {
...
+        this.publisher = checkNotNull(publisher);
     }
 
     public String getYear() {
         return year;
     }
 
     public void setYear(String year) {
-        this.year = year;
+        Preconditions.checkArgument(year.matches("[0-9]+"), "The year field must contain only numbers");
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java:110]: Nice!! :D

File: src/main/java/wolox/training/models/User.java [hunk 2]
```diff
@@ -53,31 +55,34 @@ public String getUsername() {
...
+        this.username = checkNotNull(username);
     }
 
     public String getName() {
         return name;
     }
 
     public void setName(String name) {
-        this.name = name;
+        Preconditions.checkArgument(name.chars().allMatch(Character::isLetter),
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/User.java:67]: tenor

**Review by Jimenarosello — APPROVED** [review]
Excellent job!! 😄

**Review by julianvasquez-wolox — APPROVED** [review]

**Pull request merged**